keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/artifacts/SpinnakerArtifact.kt

```diff
@@ -0,0 +1,20 @@
+package com.netflix.spinnaker.keel.api.artifacts
+
+/**
+ * An immutable data class that mirrors com.netflix.spinnaker.kork.artifacts.model.Artifact, but without
+ * all the Jackson baggage. One notable difference from the kork counterpart is that this class enforces
+ * non-nullability of a few key fields without which it doesn't make sense for an artifact to exist in
+ * Managed Delivery terms.
+ */
+data class SpinnakerArtifact(
```

Comment:
I've debated with myself about the name of this class. My second candidate would be `VersionedArtifact`, to convey the idea that the most valuable piece of information we're looking for in it is the artifact version (along with the type and name to correlate with delivery configs). Let me know what you prefer.

Reply:
I went back and forth on this as I read this PR. I think I prefer `VersionedArtifact`, but possibly even `VersionedSpinnakerArtifact` to differentiate it from, say, keel's `CompleteVersionedArtifact` .

Reply:
I think `SpinnakerArtifact` is fine, because it's a class that is pulled from the rest of spinnaker and is just used to receive data.

Reply:
Thank you both. I'll leave it as is for now, and we can revisit later.